src/PostComment.jsx

```diff
@@ -0,0 +1,95 @@
+import { postComment } from "./UTILS/utils";
+import UsernameContext from "./CONTEXTS/UsernameContext";
+import { useContext } from "react";
+import { useState } from "react";
+import { getCommentsByID } from "./UTILS/utils";
+import { getArticleByID } from "./UTILS/utils";
+
+export default function PostComment({
+  articleID,
+  setArticleComments,
+  setArticleData,
+}) {
+  const { currentUser } = useContext(UsernameContext);
+  const [commentPosted, setCommentPosted] = useState(false);
+  const [errorStatus, setErrorStatus] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [userComment, setUserComment] = useState({
+    body: "",
+    username: currentUser.username,
+  });
+
+  function handleSubmit(event) {
+    setErrorStatus(false);
+    setCommentPosted(false);
+    event.preventDefault();
+    setIsLoading(true);
+    postComment(userComment, articleID)
+      .then((response) => {
+        setUserComment({
+          body: "",
+          username: currentUser,
+        });
+        setCommentPosted(true);
+        setErrorStatus(false);
+        setIsLoading(false);
+        getCommentsByID(articleID).then((response) => {
+          setArticleComments(response.comments);
+
+          getArticleByID(articleID).then((response) => {
+            setArticleData(response.article);
```

Comment:
This seems like redundant code? If you are trying to get the comment to render please render it optimistically and then use a dependancy array to trigger a re fetching of the comments data